data-generation/src/lib.rs

```diff
@@ -0,0 +1,281 @@
+    pyarrow::PyArrowType,
+};
+use pyo3::{exceptions::PyRuntimeError, prelude::*};
+use rand::{
+    distributions::{Distribution, Standard, Uniform},
+    Rng, SeedableRng,
+};
+use rand_chacha::ChaCha8Rng;
+
+/// Parameters of transactions amount distribution
+const MIN: f64 = 100.0;
+const MAX: f64 = 10000.0;
+
+/// Parameters of transactions amount distribution
+/// We will sample in a way like from V * BOTTOM / 100 to V * UPPER / 100
+const BOTTOM: i64 = 90;
+const UPPER: i64 = 110;
+
+/// We have two kinds of cards, debit and credit
+/// We will choose DS with probability 0.75 and CC otherwise
+const CARD_TYPES: (&str, &str) = ("DC", "CC");
+
+/// We have different transactions categories
+/// We will choose them uniformly
+/// Inspired by: https://medium.com/@swedbank.tech/how-does-categorization-of-transactions-work-4262d720fd2d
+const TRANSACTION_TYPES: [&'static str; 13] = [
+    "food-and-household",
+    "home",
+    "uncategorized",
+    "leisure-and-lifestyle",
+    "health-and-beauty",
+    "shopping-and-services",
+    "children",
+    "vacation-and-travel",
+    "education",
+    "insurance",
+    "investments-and-savings",
+    "expenses-and-other",
+    "cars-and-transportation",
+];
+
+/// We have different channels
+/// We will choose mobile with probability 0.25 classic otherwise
+const CHANNELS: (&str, &str) = ("mobile", "web");
+
+/// Generate data for a single customer;
+/// Returns expected_amnt of rows in a fixed schema.
+fn generate_customer_transactions(
+    id: i64,
+    expected_amnt: i64,
+    distr_trx: Uniform<f64>,
+    distr_cat: Uniform<i64>,
+    seed: u64,
+    d_minus: i64,
+) -> (
+    Vec<i64>,
+    Vec<&'static str>,
+    Vec<&'static str>,
+    Vec<&'static str>,
+    Vec<f64>,
+    Vec<i64>,
+) {
+    let mut rng = ChaCha8Rng::seed_from_u64(seed);
+
+    // We will sample from expected amount;
+    // Interval is defined on the top of the file
+    let distr_90p = Uniform::<i64>::try_from((expected_amnt * BOTTOM / 100)..(expected_amnt * UPPER / 100)).unwrap();
+    let sampled_expected = distr_90p.sample(&mut rng);
+
+    // I will store rows as columns in form of vectors
+    let ids: Vec<i64> = vec![id; sampled_expected.as_usize()];
+    let mut card_t: Vec<&'static str> = Vec::with_capacity(sampled_expected.as_usize());
+    let mut trx_t: Vec<&'static str> = Vec::with_capacity(sampled_expected.as_usize());
+    let mut ch_t: Vec<&'static str> = Vec::with_capacity(sampled_expected.as_usize());
+    let mut amnt: Vec<f64> = Vec::with_capacity(sampled_expected.as_usize());
+    let t_minus: Vec<i64> = vec![d_minus; sampled_expected.as_usize()];
+
+
+    for _i in 0..sampled_expected {
+        // Choose DC with probability 75%
+        let card_type = if rng.gen_bool(0.75) {
+            CARD_TYPES.0
+        } else {
+            CARD_TYPES.1
+        };
+
+        // Choose random transaction type
+        let trx_type = TRANSACTION_TYPES
+            .get(distr_cat.sample(&mut rng).as_usize())
```

Comment:
`rand` crate can sample more nicely. Not sure, maybe this logic is sophisticated and can't be done by `rand`.